app/controllers/checkout_controller.rb

```diff
@@ -58,6 +61,23 @@ def update
 
   private
 
+  def set_default_bill_address
+    if params[:order][:default_bill_address] == 'YES'
+      new_bill_address = @order.bill_address.clone
+      spree_current_user.set_bill_address(new_bill_address)
+      @order.customer.update_attribute(:bill_address, new_bill_address) unless @order.customer.bill_address
+    end
+
+  end
+
+  def set_default_ship_address
+    if params[:order][:default_ship_address] == 'YES'
+      new_ship_address = @order.ship_address.clone
+      spree_current_user.set_ship_address(new_ship_address)
+      @order.customer.update_attribute(:ship_address, new_ship_address) unless @order.customer.ship_address
```

Comment:
Logic here should be to update customer's bill address if it is already set to something, but otherwise (ie. if it is blank) leave it blank. So we need to change `unless` here to `if`.